### PR TITLE
Simplify contract manager

### LINKED ISSUE #125: Importing contract_manager shouldn't have side effects
Currently importing `raiden_contracts.contract_manager` will cause a global `ContractManager` to be instantiated (which is fine) but this will in turn (depending on the (non-)existence of the precompiled contracts json archive) invoke `solc` to compile the contracts.

This shouldn't happen on import time but rather by either explicitly calling a method or implicitly on first access for example.

### PR DESCRIPTION
**Pull request by ulope**
This is a small refactor of the `ContractManager` in preparation for raiden-network/raiden#1550. Also fixes #125.

### CODE REVIEW

File: raiden_contracts/contract_manager.py [hunk 1]
```diff
@@ -1,131 +1,106 @@
...
-                files,
-                output_values=('abi', 'bin', 'ast'),
-                import_remappings=map_dirs,
-                optimize=False,
-            )
-            return fix_contract_key_names(res)
+            for contracts_dir in self._contracts_source_dirs.values():
+                res = compile_files(
+                    [str(file) for file in contracts_dir.glob('*.sol')],
+                    output_values=('abi', 'bin', 'ast'),
```
[the remark this answers is not in the document]
Comment (pcppcp) [reply raiden_contracts/contract_manager.py]: not sure, maybe some weird `solc` behavior that required enabling this option.

[the remark this answers is not in the document]
Comment (palango) [reply raiden_contracts/contract_manager.py]: @ulope Can you check if it works without `ast`?

File: raiden_contracts/contract_manager.py [hunk 2]
```diff
@@ -1,131 +1,106 @@
...
+                    import_remappings=import_dir_map,
+                    optimize=False,
+                )
+                self._contracts.update(_fix_contract_key_names(res))
         except FileNotFoundError:
-            raise Exception('Could not compile the contract. Check that solc is available.')
+            raise RuntimeError(
+                'Could not compile the contract. Check that solc is available.',
+            ) from FileNotFoundError
```
[the remark this answers is not in the document]
Comment (ulope) [reply raiden_contracts/contract_manager.py]: Err, yeah... thanks for noticing.

File: raiden_contracts/contract_manager.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ulope) [reply raiden_contracts/contract_manager.py]: Will do

[the remark this answers is not in the document]
Comment (ulope) [reply raiden_contracts/contract_manager.py]: Yep, it's 'required' (i.e. crashes when `ast` isn't included) by py-solc. 
[Filed an issue for it.](https://github.com/ethereum/py-solc/issues/56)

**Pull request merged**